### ISSUE TITLE
Empty array wrongly treat as string

### ISSUE THREAD

**Issue by andrAnat-b** [crownedgrouse/jason#4]
I had try to run my example and find some mistake:
`Json = #{allowed_updates => [],limit => 1,offset => 1,timeout => 30}}.`
`jason:encode(JSON,[{mode, map}, {binary, kv}]).                                                                                    
"{\"allowed_updates\": \"\",\"limit\": 1,\"offset\": 1,\"timeout\": 30}"`

allowed_updates transform [] to ""

Please help me to understand - can I strict this behavior and treat all lists as arrays. Thank You.

**Comment by crownedgrouse** [comment]
Hi,
[] is not an array, it is an Erlang list.
It is also an empty string in Erlang (an empty list of characters ).
[] =:= ""

**Comment by crownedgrouse** [comment]
Note : binary option has no action for encode.
It is aimed for decoding a JSON to an Erlang term.

**Comment by crownedgrouse** [comment]
Array in Erlang is an oddity, few used : https://erlang.org/doc/man/array.html

Jason do not handle this, but could be nice to describe json arrays. Especially for empty list.

I will consider this.

**Comment by andrAnat-b** [comment]
> Note : binary option has no action for encode. It is aimed for decoding a JSON to an Erlang term.

Thank you I understand it.

**Comment by andrAnat-b** [comment]
> Hi, [] is not an array, it is an Erlang list. It is also an empty string in Erlang (an empty list of characters ). [] =:= ""

I agree with you. But need some type of representation of JSON array for erlang terms. Other parsers treat lists as JSON arrays except lists with tuples on two elements. I was confused when I'm tried to produce JSON array from list [97, 98] and got string  "ab" :)

**Comment by crownedgrouse** [comment]
please tell me which parser you are referring to ? To understand behavior you were expecting.
[97,98] is the same than "ab" in erlang shell display, but you application using result could use it as a list of integers.

**Comment by andrAnat-b** [comment]
I compare behavior with jsx and jsone. They treat lists as arrays and binaries as strings.
